src/grz_cli/commands/clean.py

```diff
@@ -0,0 +1,78 @@
+"""Command for cleaning a submission from the S3 inbox."""
+
+import logging
+
+import click
+
+from ..transfer import init_s3_resource
+from ..utils.config import read_config
+from .common import config_file, submission_id
+
+log = logging.getLogger(__name__)
+
+
+@click.command()
+@submission_id
+@config_file
+@click.option("--yes-i-really-mean-it", is_flag=True)
+def clean(submission_id, config_file, yes_i_really_mean_it: bool):
+    """
+    Remove all files of a submission from the S3 inbox.
+    """
+    config = read_config(config_file)
+    bucket_name = config.s3_options.bucket
+
+    if yes_i_really_mean_it or click.confirm(
+        f"Are you SURE you want to delete the submission from the inbox bucket ({bucket_name})?",
+        default=False,
+        show_default=True,
+    ):
+        prefix = submission_id
+        prefix = prefix + "/" if not prefix.endswith("/") else prefix
+
+        resource = init_s3_resource(config)
+        bucket = resource.Bucket(bucket_name)
+        log.info(f"Deleting {prefix} from {bucket_name} …")
+
+        responses = bucket.objects.filter(Prefix=prefix).delete()
+        if not responses:
+            raise ValueError(f"No objects with prefix {prefix} in bucket {bucket_name} found for deletion.")
```

Comment:
This is a bit nitpicky for the CLI of an internal tool, but I'd use [`click.Context.fail`](https://click.palletsprojects.com/en/stable/api/#click.Context.fail) or `sys.exit(1)` here instead of raising an exception that currently isn't caught and makes a big traceback. LGTM otherwise!

Reply:
I think sys.exit is the way to go, click's fail prints the help by default etc, so is more related to the actual CLI part failing.